### ISSUE TITLE
some problems about MNIST-Keras

### ISSUE THREAD

**Issue by zyf654321** [HewlettPackard/swarm-learning#42]
Why is there a live sn node after deploying the SN node through the following command
![image](https://user-images.githubusercontent.com/88707387/128830343-69df130e-8e8f-4c1c-af70-b1b560e360a7.png)
![image](https://user-images.githubusercontent.com/88707387/128830383-a314f251-c1e4-4d4d-bf8d-fa6d28215f24.png)
After the SL node is deployed through the following command, the number of SN nodes of live becomes 0.  Why ?
![image](https://user-images.githubusercontent.com/88707387/128830414-a9f1b450-959b-4220-9a02-bd8ef43857c4.png)
![image](https://user-images.githubusercontent.com/88707387/128830422-d88e1721-21fc-41cf-ba9e-ada7f3170e13.png)
And the following errors occurred, resulting in running failure.
![image](https://user-images.githubusercontent.com/88707387/128830454-0232d3a8-fbbc-43cf-a03a-717a589f33a1.png)
![image](https://user-images.githubusercontent.com/88707387/128830471-af071a19-c8cb-4fcf-99c6-18be5cfa680a.png)

**Comment by RadhakrishnaJ** [comment]
SN became live after running init_workspace, that is expected.  Some error has happened after that which killed SN node.
As there no SN node, run-sl does not initiate Swarm Learning process.

Do you have access to use Docker group in this machine? 
Are you using single machine to run all the components? if using multiple machines, make sure they are accessible from one another.

Can you delete work space, and start a clean execution from init once again?
If same error exists then send full screenshot of run-sl command and docker logs of SN container.

**Comment by zyf654321** [comment]
Sorry to bother you again. I solved the problem. After node1 for SL is deployed, the following information is output.
![52981fa728dd2ef739fc00c3a36b927](https://user-images.githubusercontent.com/88707387/129149164-8d3057e3-8d50-4037-9566-ff6d244e3617.png)
Is it true that I just need to wait for the model training to be completed?(I'm not using a GPU)

Now, the problem is, after I've waited a long time, the program doesn't have any other output.The output shows that it has been "Merging..."State,why is that?

**Comment by RadhakrishnaJ** [comment]
SL training is about to start. But it seems it has hit by memory limitation already, check your resources availability after this step.

You should see Epoch starting after merging like below - 
2021-08-11 14:30:43,344 : swarm.smlCBB : INFO :     adaptive_sync     : False
2021-08-11 14:30:43,344 : swarm.smlCBB : INFO : ***************************
2021-08-11 14:30:44,036 : swarm.mlCnt : INFO : Merging ...
Epoch 1/5

  1/469 [..............................] - ETA: 0s - loss: 2.3042 - accuracy: 0.0469
  2/469 [..............................] - ETA: 22s - loss: 2.2997 - accuracy: 0.1016

**Comment by zyf654321** [comment]
Will these warnings affect the training of the model?thank you.
![image](https://user-images.githubusercontent.com/88707387/129194122-73dde4de-133d-469b-808a-0148a1553c26.png)

**Comment by RadhakrishnaJ** [comment]
> Will these warnings affect the training of the model?thank you.
> ![image](https://user-images.githubusercontent.com/88707387/129194122-73dde4de-133d-469b-808a-0148a1553c26.png)

This does not impact training.  You should have memory and computing resources available to run training.

**Comment by zyf654321** [comment]
Ok, so I should clean up the system memory.Could you tell me roughly how much space the data set used for training takes up?In addition, I want to use COVID-19 data for training, how should I set it up?

**Comment by RadhakrishnaJ** [comment]
> Ok, so I should clean up the system memory.Could you tell me roughly how much space the data set used for training takes up?In addition, I want to use COVID-19 data for training, how should I set it up?

Our testing happened in 32GB RAM machines. Not sure what is the minimum required here. 
My suggestion is observe your resources through task manager (memory and swap memory), if you hit upper boundary on resources then training might be very slow or you might get some timeout errors.
You can update example code read subset of mnist data, instead of 60k try with 10K samples as start to reduce memory load on training. If 10K training is successful then see if you can increase training data further.

**Comment by RadhakrishnaJ** [comment]
As there is no issue with Swarm, closing this issue now. Requesting user to raise new issue for any further support.
